pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,75 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet
+type SubnetSpec struct {
+	// Size of subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPV4SubnetSize string `json:"ipv4SubnetSize"`
+	// Access mode of subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	AccessMode string `json:"accessMode"`
+	// Subnet CIDR.
+	IPAddresses string `json:"ipAddresses,omitempty"`
```

Comment:
From NSX-T VPC subnet yaml, this field is an array with maxItems:2. Could you try the nsx subnet API that if we can specify 2 cidrs? If yes, we can add the below validations.
        // +kubebuilder:validation:MinItems=0
	// +kubebuilder:validation:MaxItems=2
ip_addresses:
      type: array
      maxItems: 2
      items:
        type: string
        format: ip-cidr-block
      title: CIDR

Reply:
Thanks and fixed.